CTFd/api/v1/challenges.py

```diff
@@ -376,25 +418,19 @@ def get(self, challenge_id):
 
         response = chal_class.read(challenge=chal)
 
-        Model = get_model()
-
         if scores_visible() is True and accounts_visible() is True:
-            solves = Solves.query.join(Model, Solves.account_id == Model.id).filter(
-                Solves.challenge_id == chal.id,
-                Model.banned == False,
-                Model.hidden == False,
+            solves_q = _build_solves_query(
+                admin_view=is_admin(), extra_filters=(Solves.challenge_id == chal.id,)
             )
-
-            # Only show solves that happened before freeze time if configured
-            freeze = get_config("freeze")
-            if not is_admin() and freeze:
-                solves = solves.filter(Solves.date < unix_time_to_utc(freeze))
-
-            solves = solves.count()
-            response["solves"] = solves
+            # If there are no solves for this challenge ID then we have 0 rows
+            maybe_row = solves_q.first()
+            if maybe_row:
+                _, solve_count, solved_by_user = maybe_row
+                solved_by_user = bool(solved_by_user)
+            else:
+                solve_count, solved_by_user = 0, False
         else:
-            response["solves"] = None
-            solves = None
+            solve_count, solved_by_user = None, False
```

Comment:
I feel as if this line is wrong? scores_visible and accounts_visible wouldn't affect whether the user can see if their own challenge is solved?

I haven't tested it myself yet though.

Reply:
Yeah solved_by_user should be independent of scores_visible() and accounts_visible() and always point out whether the user solved their challenge.

`/api/v1/users/me/solves`
```json
{
            "type": "correct",
            "challenge_id": 6,
            "user": 2,
            "team": 7,
            "challenge": {
                "value": 150,
                "category": "Forensics",
                "name": "light"
            },
            "date": "2021-02-25T10:00:37.041914+00:00",
            "id": 60
        }
```

compared to:

`/api/v1/challenges/6`
```json
        "solves": null,
        "solved_by_me": false,
```

Reply:
Sure thing, that sounds entirely reasonable. So we'd want the response to say this following in that case?
```
        "solves": null,
        "solved_by_me": true,
```

Reply:
It would be dependent on whether the user actually solved the challenge. In the example data I showed above it would say true. 

You probably need to move the solve_q variable out of the if statement and then refer to it from within.